File: ichecknetwork/src/main/java/vn/icheck/android/network/base/ICNetworkCallbackManager.java
package vn.icheck.android.network.base;

import okhttp3.Interceptor;

public interface ICNetworkCallbackManager {


    void onRequiredLogin(int code, Interceptor.Chain chain);

    void onLoginSuccess(int code);

    void onNetworkError(Throwable throwable);

    /**
     * The factory class for the {@link ICNetworkCallbackManager}.
     */
    class Factory {
        /**
         * Creates an instance of {@link ICNetworkCallbackManager}.
         *
         * @return an instance of {@link ICNetworkCallbackManager}.
         */
        public static ICNetworkCallbackManager create() {
            return new CallbackManagerImpl();
        }
    }
}

File: icheckloyalty/src/main/java/vn/icheck/android/loyalty/network/ICNetworkCallbackManager.kt
package vn.icheck.android.loyalty.network

import okhttp3.Interceptor

interface ICNetworkCallbackManager {

    fun onRequiredLogin(code: Int, chain: Interceptor.Chain?)

    fun onLoginSuccess(code: Int)

    fun onNetworkError(throwable: Throwable?)

    object Factory {
        /**
         * Creates an instance of [ICNetworkCallbackManager].
         *
         * @return an instance of [ICNetworkCallbackManager].
         */
        fun create(): ICNetworkCallbackManager {
            return CallbackManagerImpl()
        }
    }
}